CHANGELOG.md

```diff
@@ -1,6 +1,9 @@
 ## HEAD
 
 * DHCP support. [GH-51]
+* Non-A-queries (like IPv6 AAAA) matching a configured pattern will be answered with a `NOTIMP` error. [GH-76]
```

Comment:
If we keep the pass-through functionality I think we should forward these to the pass-through server as well? That way we won't break *any* previous usage.

Basically I'm thinking (pseudo code):

```
if non-matching-domain or record-type != A:
    if VagrantDNS::Config.resolver != false:
        send-verbatim-upstream(query)
    elseif non-matching-domain
        return NXDOMAIN
    else # record-type != A
        return NOTIMP
    fi
fi
```

Reply:
I think both are two valid use-cases:
1. My local host names are not mirrored publicly. If any non-A query matching a configured pattern, should be NXDomain
2. My local host names "overwrite" public DNS.  Any query not matching any configured pattern should be passed through

Maybe it's easier to have a dedicated `passthrough` setting. That way we could passthrough:
- `true`: everything that didn't match
- `false`: disable passthrough. known, non-A queries will fail  with NotImp 
- `:unknown`: only unknown patterns. known, non-A queries will fail with NotImp

Reply:
While the `true` option matches the current behavior, I think `:unknown` or even `false` is what most people would want to set.